core/src/main/java/org/gbif/pipelines/core/ws/client/internal/GBIFInternalServiceClient.java

```diff
@@ -0,0 +1,174 @@
+package org.gbif.pipelines.core.ws.client.internal;
+
+import org.gbif.pipelines.core.ws.HttpClientFactory;
+import org.gbif.pipelines.core.ws.HttpConfigFactory;
+import org.gbif.pipelines.core.ws.config.Config;
+import org.gbif.pipelines.core.ws.config.Service;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import okhttp3.OkHttpClient;
+import retrofit2.Call;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+/**
+ * rest client for getting gbif internal api responses
+ */
+public class GBIFInternalServiceClient {
+
+  private static final String INSTALLATION_KEY = "installationKey";
+  private static final String PUB_ORGANIZATION_KEY = "publishingOrganizationKey";
+  private static final String DATASET_TITLE_KEY = "title";
+  private static final String PUB_ORG_COUNTRY_KEY = "country";
+  private static final String INSTALLATION_TYPE_KEY = "type";
+  private static final String DATASET_NETWORK_KEY = "key";
+  private final GBIFInternalService service;
+  //caches the last requests obtained
+  private final Cache<String, GBIFInternalResponse> datasetResponseCache =
```

Comment:
Maybe we should remove the Guava Cache from here and delete that cross cutting concern to the user of this class, all this client can be easily wrapped around a Memoizer like the one proposed in a different branch